Added SensorMsgImage_CvMat.hpp/.cpp

AntoBongio opened the pull request:
Implements #1 

NOTE:
- Check name convention
- Consider removing cv_bridge
- This implementation is constrained to the following sensor_msgs::image_encodings: {MONO8, BGR8, MONO16, BHR16, BGRA8, BGRA16}

AntoBongio committed:
Added SensorMsgImage_CvMat.hpp/.cpp

NOTE:
- Check name convention
- Consider removing cv_bridge
- This implementation is constrained to the following sensor_msgs::image_encodings: {MONO8, BGR8, MONO16, BHR16, BGRA8, BGRA16}

roncapat committed:
Trigger for workflow

roncapat commented on src/SensorMsgsImage_CvMat.cpp:
```diff
@@ -0,0 +1,70 @@
+    std::cerr << "[SensorMsgsImage_CvMat Adapter] Conversion from message" << std::endl;
+
+    destination.header = source.header;
+    destination.mat = cv_bridge::toCvCopy(source, source.encoding)->image;
```
Forse qua si può fare "move" per evitare la copia dei dati dalla `->image` alla `destination.mat`

roncapat replied:
cv::Mat supporta la move in teoria (https://docs.opencv.org/4.x/d3/d63/classcv_1_1Mat.html#adcaeb226bafeff8a1b262f57f6ffbcc4)

AntoBongio replied:
Si, potevo farla. Ho sistemato.

AntoBongio committed:
Added move for cv::Mat.

AntoBongio committed:
Merge branch 'main' into bongio_pull

roncapat commented:
## Proposal:

### Structure names:

StampedPointCloud_PCL\
StampedImage_CV\
StampedOccupancyGrid_CV

### Headers:

SensorsMsgImage_CvMat.hpp\
NavigationOccupancyGrid_CvMat.hpp\
SensorMsgPointCloud2_PCLPointCloud.hpp

### Dummy headers:

They simply include above headers, grouped by support library used. The user should include these headers directly instead of the ones above.

CV.hpp\
PCL.hpp

AntoBongio committed:
Refactor.

Structure names:
StampedPointCloud2 ---> StampedPointCloud_PCL
StampedCvMat ---> StampedOccupancyGrid_CV
CustomCvMat ---> StampedImage_CV

Headers name convention changed.

Added dummy headers to collect all the adapters related to the same "type" of structure.

Fixit-Davide committed:
Last refactor round
